src/service/viewport-impl.js

```diff
@@ -140,6 +143,16 @@ export class Viewport {
 
     this.binding_.onScroll(this.scroll_.bind(this));
     this.binding_.onResize(this.resize_.bind(this));
+
+    this.onScroll(() => {
+      if (!this.duringAnimationFrame_) {
+        this.duringAnimationFrame_ = true;
+        this.vsync_.measure(() => {
+          this.duringAnimationFrame_ = false;
+          this.viewer_.postScroll(this.binding_.getScrollTop());
```

Comment:
> Isn't that enough?

Not necessarily, this can cause another style recalc, which isn't needed since we already have it cached.

>  The scroll_() function in viewport-impl called this.binding_.getScrollTop() every time it scrolls.

That's also expensive, but required. That's actually what sets the cached `scrollTop` value.